### PR TITLE
fix: theme id and headerOptions in TemplateListPage

### LINKED ISSUE #21313: 🐛 Bug Report: ScaffolderPage does not follow headerOptions after 1.20 update
### 📜 Description

Passing headerOptions to <ScaffolderPage> does not work anymore after 1.20 update.

### 👍 Expected behavior

Page title, title and subtitle should be what set in the headerOptions

### 👎 Actual Behavior with Screenshots

Defaults are always visible from https://github.com/backstage/backstage/blob/ca4063dd2704eebc630d946bf94cb435e7efd742/plugins/scaffolder/src/next/TemplateListPage/TemplateListPage.tsx#L156

### 👟 Reproduction steps

Add headerOptions for the ScaffolderPage

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

OS:   Darwin 23.1.0 - darwin/arm64
node: v18.15.0
yarn: 1.22.19
cli:  0.24.0 (installed)
backstage:  1.20.1

Dependencies:
  @backstage/app-defaults                                            1.4.5
  @backstage/backend-app-api                                         0.5.8
  @backstage/backend-common                                          0.17.0, 0.19.9
  @backstage/backend-dev-utils                                       0.1.2
  @backstage/backend-openapi-utils                                   0.1.0
  @backstage/backend-plugin-api                                      0.6.7
  @backstage/backend-tasks                                           0.5.12
  @backstage/backend-test-utils                                      0.2.8
  @backstage/catalog-client                                          1.4.6
  @backstage/catalog-model                                           1.4.3
  @backstage/cli-common                                              0.1.13
  @backstage/cli-node                                                0.2.0
  @backstage/cli                                                     0.24.0
  @backstage/config-loader                                           1.5.3
  @backstage/config                                                  1.1.1
  @backstage/core-app-api                                            1.11.1
  @backstage/core-components                                         0.12.5, 0.13.8
  @backstage/core-plugin-api                                         1.8.0
  @backstage/dev-utils                                               1.0.23
  @backstage/errors                                                  1.2.3
  @backstage/eslint-plugin                                           0.1.3
  @backstage/frontend-plugin-api                                     0.3.0
  @backstage/integration-aws-node                                    0.1.8
  @backstage/integration-react                                       1.1.21
  @backstage/integration                                             1.7.2
  @backstage/plugin-api-docs                                         0.10.0
  @backstage/plugin-app-backend                                      0.3.55
  @backstage/plugin-app-node                                         0.1.7
  @backstage/plugin-auth-backend-module-gcp-iap-provider             0.2.1
  @backstage/plugin-auth-backend-module-github-provider              0.1.4
  @backstage/plugin-auth-backend-module-gitlab-provider              0.1.4
  @backstage/plugin-auth-backend-module-google-provider              0.1.4
  @backstage/plugin-auth-backend-module-oauth2-provider              0.1.4
  @backstage/plugin-auth-backend                                     0.20.0
  @backstage/plugin-auth-node                                        0.2.19, 0.4.1
  @backstage/plugin-catalog-backend-module-aws                       0.3.1
  @backstage/plugin-catalog-backend-module-msgraph                   0.5.14
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model   0.1.4
  @backstage/plugin-catalog-backend-module-unprocessed               0.3.4
  @backstage/plugin-catalog-backend                                  1.15.0
  @backstage/plugin-catalog-common                                   1.0.18
  @backstage/plugin-catalog-graph                                    0.3.0
  @backstage/plugin-catalog-node                                     1.5.0
  @backstage/plugin-catalog-react                                    1.9.0
  @backstage/plugin-catalog-unprocessed-entities                     0.1.5
  @backstage/plugin-catalog                                          1.15.0
  @backstage/plugin-devtools-backend                                 0.2.4
  @backstage/plugin-devtools-common                                  0.1.6
  @backstage/plugin-devtools                                         0.1.6
  @backstage/plugin-entity-feedback-backend                          0.2.4
  @backstage/plugin-entity-feedback-common                           0.1.3
  @backstage/plugin-entity-feedback                                  0.2.9
  @backstage/plugin-events-node                                      0.2.16
  @backstage/plugin-github-actions                                   0.6.7
  @backstage/plugin-home-react                                       0.1.5
  @backstage/plugin-home                                             0.5.10
  @backstage/plugin-jenkins-backend                                  0.3.1
  @backstage/plugin-jenkins-common                                   0.1.21
  @backstage/plugin-jenkins                                          0.9.1
  @backstage/plugin-kubernetes-common                                0.7.1
  @backstage/plugin-org                                              0.6.16
  @backstage/plugin-permission-backend                               0.5.30
  @backstage/plugin-permission-common                                0.7.10
  @backstage/plugin-permission-node                                  0.7.18
  @backstage/plugin-permission-react                                 0.4.17
  @backstage/plugin-proxy-backend                                    0.4.5
  @backstage/plugin-scaffolder-backend-module-confluence-to-markdown 0.2.8
  @backstage/plugin-scaffolder-backend-module-cookiecutter           0.2.31
  @backstage/plugin-scaffolder-backend                               1.19.0
  @backstage/plugin-scaffolder-common                                1.4.3
  @backstage/plugin-scaffolder-node                                  0.2.8
  @backstage/plugin-scaffolder-react                                 1.6.0
  @backstage/plugin-scaffolder                                       1.16.0
  @backstage/plugin-search-backend-module-catalog                    0.1.11
  @backstage/plugin-search-backend-module-elasticsearch              1.3.10
  @backstage/plugin-search-backend-module-techdocs                   0.1.11
  @backstage/plugin-search-backend-node                              1.2.11
  @backstage/plugin-search-backend                                   1.4.7
  @backstage/plugin-search-common                                    1.2.8
  @backstage/plugin-search-react                                     1.7.3
  @backstage/plugin-search                                           1.4.2
  @backstage/plugin-sonarqube-backend                                0.2.9
  @backstage/plugin-sonarqube-react                                  0.1.10
  @backstage/plugin-sonarqube                                        0.7.8
  @backstage/plugin-techdocs-backend                                 1.9.0
  @backstage/plugin-techdocs-node                                    1.10.0
  @backstage/plugin-techdocs-react                                   1.1.13
  @backstage/plugin-techdocs                                         1.9.0
  @backstage/plugin-user-settings                                    0.7.12
  @backstage/release-manifests                                       0.0.11
  @backstage/test-utils                                              1.4.5
  @backstage/theme                                                   0.2.19, 0.4.4
  @backstage/types                                                   1.1.1
  @backstage/version-bridge                                          1.0.7

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by fedy97**
## Hey, I just made a Pull Request!

- Added `headerOptions` to `TemplateListPage` to optionally override default values.
- Changed themeId of TemplateListPage from `website` to `home`.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### Related Issue

Closes #21313

### CODE REVIEW

File: .changeset/dry-readers-raise.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-scaffolder': patch
+---
+
+Added `headerOptions` to `TemplateListPage` to optionally override default values.
+Changed themeId of TemplateListPage from `website` to `home`.
```
Comment (awanlin) [inline .changeset/dry-readers-raise.md:6]:
```suggestion
Changed `themeId` of TemplateListPage from `website` to `home`.
```

That should make Vale happy

**Review by awanlin — CHANGES_REQUESTED** [review]
This looks good over all, just fixing some regressions so thanks for that @fedy97 👍 

Left a comment on how to fix the Vale error and you'll also want to re-generate the API Report for this package, there's some docs on this here: https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#api-reports

**Comment by fedy97** [comment]
> This looks good over all, just fixing some regressions so thanks for that @fedy97 👍
> 
> Left a comment on how to fix the Vale error and you'll also want to re-generate the API Report for this package, there's some docs on this here: https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#api-reports

thanks @awanlin

**Comment by benjdlambert** [comment]
@fedy97 thanks for this and the quick turnaround! :tada:

**Review by tudi2d — APPROVED** [review]

**Pull request merged**